core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -424,10 +424,6 @@ protected Scope visitInsert(Insert insert, Optional<Scope> scope)
             }
```

Comment:
> In case of INSERT when row is matching a row filter then query will
fail.

According to the SQL spec, it's the other way around. If the row *does not* match the row filter, then the query should fail.

> In case of DELETE and UPDATE rows that are not matching the row filters
are not read at all and so they are not modified or deleted.

Similarly, for UPDATE, if the result of applying the update is a row that does not match the filter, the query should fail. For instance, if the row filter says `WHERE x = 1`,

given a `row(x) == (1)`, the following query should fail:

```
UPDATE t SET x = 2
```

The row qualifies for consideration for the update because it matches the row filter, but the resulting row after the update is applied (i.e., `(2)`) no longer satisfies the filter.

Reply:
I am sorry, my bad. I haven't updated the commit message. It is like was initially. Fixed.

Reply:
@martint ping